### ISSUE TITLE
Revise v3.3.0 seems broken

### ISSUE THREAD

**Issue by originalsouth** [timholy/Revise.jl#664]
I have a project Dummy with the contents
```
module Dummy

function greet()
    println("Potato")
    #println("Tomato")
end

end
```

Which works as expected with Revise v3.2.1:
```
% julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.1 (2021-12-22)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |

(@v1.7) pkg> st Revise
      Status `~/.julia/environments/v1.7/Project.toml`
  [295af30f] Revise v3.2.1 `https://github.com/timholy/Revise.jl.git#v3.2.1`

> using Revise

> using Dummy
[ Info: Precompiling Dummy [cbdcda1a-e039-4a41-9910-8e68b339a203]

> Dummy.greet()
Potato

> Dummy.greet() # Swap the comment hash for the println statement
Tomato
```

Yet with Revise v3.3.0 I obtain:
```
% julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.1 (2021-12-22)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |

(@v1.7) pkg> st Revise
      Status `~/.julia/environments/v1.7/Project.toml`
  [295af30f] Revise v3.3.0

> using Revise

> using Dummy
[ Info: Precompiling Dummy [cbdcda1a-e039-4a41-9910-8e68b339a203]

> Dummy.greet()
Potato

> Dummy.greet() # Swap the comment hash for the println statement
Potato
```
So for now I am using:
```
(@v1.7) pkg> add Revise#v3.2.1
```

**Comment by fingolfin** [comment]
Just came here to report the same but with Julia 1.6.4

**Comment by ronisbr** [comment]
Same problem here with Julia 1.7.1.

**Comment by xijiang** [comment]
Same here. I downgrade it to 3.2.1 to continue my work.

**Comment by timholy** [comment]
Very interesting. For me it works if you load Revise in your `~/.julia/config/startup.jl` (which is how I always do it) but not if you load it manually in your session. I will investigate tomorrow.

**Comment by adannenberg** [comment]
If you have 3.3.0 added and you load Revise in your startup.jl file then Revise recognizes changes to strings (e.g. print("Hello") versus print("Hello2")) but it still doesn't recognize when export statements are commented out / uncommented.  3.2.1 does not have this problem

**Comment by originalsouth** [comment]
> Very interesting. For me it works if you load Revise in your `~/.julia/config/startup.jl` (which is how I always do it) but not if you load it manually in your session. I will investigate tomorrow.

Confirming that for me indeed the issue is resolved by placing `using Revise` in `~/.julia/config/startup.jl`.

**Comment by Wikunia** [comment]
Same problem for me and just to add one thing: running `revise()` "fixes" it. Meaning it reloads it correctly for that one moment but as expected after another change it fails again.

**Comment by timholy** [comment]
> but it still doesn't recognize when export statements are commented out / uncommented. 3.2.1 does not have this problem

Can you supply a reproducer? Revise has never, to my knowledge, been able to "walk back" bindings. You can add them though.

**Comment by adannenberg** [comment]
You're exactly right.  Revise recognized uncommenting an export statement,
but recommenting the (already executed) export statement had no effect.
Sorry if I spread my confusion to others :<
_______________________
Alex Dannenberg
33 The Ledges Road
Newton MA 02459
Tel  617.584.8969
***@***.***

**Comment by timholy** [comment]
> running revise() "fixes" it

Yeah, it was just a problem in automatically running `revise()` each time you enter a command at the REPL. Revise does that by adding to a REPL's `ast_transformations`, but the correct initialization logic was accidentally eliminated in 47c73ed99b24dfad3440cde1d5d83d6a05ef459c.

Fixed in #665 and Revise 3.3.1.
